File: Sistema para Bibliotecas Públicas/Gerenciador Biblioteca/Biblioteca/Models/Livro.cs
namespace Biblioteca.Models
{
    public class Livro
    {
        public Livro()
        {
            LivroId = Guid.NewGuid().ToString();
        }

        public string? LivroId { get; set; }

        public string? Titulo { get; set; }

        public string? Genero { get; set; }

        public int QtdExemplares { get; set; }

        public DateTime CriadoEm { get; set; } = DateTime.Now;
        
        public int AnoLancamento { get; set; }

        public string? Editora { get; set; }

        public Autor? Autor { get; set; }

        public string AutorId { get; set; }
    }
}

File: Sistema para Bibliotecas Públicas/my-app/src/models/Livro.ts
import internal from "stream"
import { Autor } from "./Autor"
import { Emprestimo } from "./Emprestimo";

export interface Livro {
    livroId?: string;
    emprestimoId?: string;
    titulo: string;
    genero: string;
    qtdExemplares: number;
    anoLancamento: number;
    criadoEm?: string;
    editora: string
    autorId: string;
    autor?: Autor; 
    emprestimo?: Emprestimo;
}